### ISSUE TITLE
Breaking change v14 => v15: formulas syntax and documentation

### ISSUE THREAD

**Issue by eXistenZe** [2sic/2sxc#3055]
### I'm submitting a ...
[x] support request

### ...about
[x] Content Types or data management
[x] APIs like REST
[x] other / unknown

### Current Behavior / Expected Behavior
In v14, I used the following formula to get data from a custom API and filling notes or other stuff:

```
v1(data, context) {
  if (data.Intervention == null || data.Intervention == "") return "";

  if (data.Intervention != context.cache.Intervention) {
    context.cache.Intervention= data.Intervention;
    delete(context.cache.data);
    context.cache.alreadyRun = false;
  }
  if (context.cache.data) return context.cache.data;  
  if (context.cache.alreadyRun) return data.value;
  context.cache.alreadyRun = true;

  context.sxc.webApi.fetchJson('app/auto/api/ReturnIntSpecs/IntProcedure?Intervention=' + data.Intervention)
    .then(data => {
      console.log('got data', data);
      context.cache.data = data;
      context.form.runFormulas();
    });
}
```

After updating to v15, the form broke and I get the error:
V1 formulas accept only simple values in return statements. If you need to return an complex object, use V2 formulas.

After changing to 
`v2((data, context) => {`
I now get the error:
form.runFormulas() is not supported in V2 formulas. Just return the promise. Formulas will auto-run when it completes.

First things first, this is a significant breaking change with no mention of it in the git release notes.
Even after realizing it, the documentation on the feature changes is nearly non-existent.
I had to look at a couple pages just to find the need to enter debug mode, and where exactly the new menu was. The live edit form was the last place anyone would look to change any sort of setting.

Please place huge alerts for breaking changes on any update.
Please keep the old functionalities running while people adapt to one huge change at a time.
Please document every code update needed to keep current apps working.

**Comment by iJungleboy** [comment]
Existing formulas should NOT break. 
We're introducing new features, but you should not have to change anything.

@DujeVatavuk pls have a quick look to see why v1 showed that error.
My guess is that the formula doesn't return anything - so maybe v1 is handling undefined incorrectly?

**Comment by iJungleboy** [comment]
@eXistenZe what data is your formula returning? a string, number, or something more complex? or `undefined` (which should be ok)?

**Comment by eXistenZe** [comment]
In this case a string.

**Comment by eXistenZe** [comment]
I apologize for the oversight, but there is another error preceding the one mentioned above:
https://site/api/2sxc/app/auto/api/ReturnIntSpecs/IntSpecsData?Intervention=something&Group=1&appId=5&zoneId=2
results in
500 Internal Server Error
Message  "An error has occurred."
This is not the usual compilation error since nothing is logged at the dnn admin logs or server logs.

This is the code on the API side, if it helps:

```
    [HttpGet]
    public string IntProcedure(string Intervention)
    {
        var procedureData = "Sem procedimento";

        var getInts = AsList(App.Data["StageTwoInterventions"]);
        var selectedInt = getInts.Where( i => i.CatId == Intervention);

        if(selectedInt.Count() == 1) {
            procedureData = selectedInt.First().Procedure.Replace("(HM)", "<img src=\"" + App.Resources.hm + "\" alt=\"hm\" width=\"30\" height=\"30\">");
        }

        return procedureData;
    }
```

**Comment by eXistenZe** [comment]
Yep. Fixed in 15.06.
The request remains nevertheless... Please document v2 formula changes (both the code changes and the settings location).
I really really recommend going back on the decision to place formulas in the debug menu. It's simple just not the place to change settings.

**Comment by iJungleboy** [comment]
Using v2 has a lot of really awesome features such as promise support, IntelliSense, stopping the formula after a certain condition and more.

We're still fine-tuning it so we'll probably officially release and document it in a few weeks.

What do you mean by "placing the formulas int the debug menu"?

**Comment by eXistenZe** [comment]
As far as I understood, editing formulas now needs access to the edit form debug menu (ctrl alt shift click).

**Comment by iJungleboy** [comment]
This is correct. It has a few reasons:

1. On one hand you need to develop it on a real form, so you see what happens
2. The form also knows about more fields etc.
3. and you can debug what you're doing in real time

The debug mode was renamed to developer mode BTW. 

Sidenote: the developer mode can only be activated by super users (unless a feature is toggled), so this is to prevent normal users from doing this.

So basically all this is by design, I don't think we should change this.
